src/components/CohortBuilder/ParticipantsTableView/index.js

```diff
@@ -14,33 +15,88 @@ const enhance = compose(
   withTheme,
   withState('pageSize', 'setPageSize', 10),
   withState('pageIndex', 'setPageIndex', 0),
+  withState('selectedRows', 'setSelectedRows', []),
+  withState('allRowsSelected', 'setAllRowsSelected', false),
 );
 
-const ParticipantsTableView = ({ sqon, api, pageIndex, pageSize, setPageIndex, setPageSize }) => (
-  <QueriesResolver
-    name="GQL_PARTICIPANTS_TABLE"
-    api={api}
-    sqon={sqon}
-    queries={[participantsQuery(sqon, pageSize, pageIndex)]}
-  >
-    {({ isLoading, data, error }) =>
-      error ? (
-        <TableErrorView error={error} />
-      ) : (
-        <React.Fragment>
+const ParticipantsTableView = ({
+  sqon,
+  onRemoveFromCohort,
+  api,
+  pageIndex,
+  setPageIndex,
+  pageSize,
+  setPageSize,
+  selectedRows,
+  setSelectedRows,
+  allRowsSelected,
+  setAllRowsSelected,
+}) => {
+  return (
+    <QueriesResolver
+      name="GQL_PARTICIPANTS_TABLE"
+      api={api}
+      queries={[participantsQuery(sqon, pageSize, pageIndex)]}
+    >
+      {({ isLoading, data, error }) => {
+        if (error) {
+          return <TableErrorView error={error} />;
+        }
+
+        const isRowSelected = node =>
+          allRowsSelected || selectedRows.some(row => row === node.participantId);
```

Comment:
So looks like we are currently not allowing removal of rows from the list once `allRowsSelected` is true. 

I am ok with moving this forward as is regarding this point, but since we have disabled removal of everything from the cohort, it would make sense to allow such use case, else I don't think the use of sets is well justified here. Maybe we would rather allow removal of sets from cohort and allow user to modify the sqon after the fact?

@rosibaj this is something we should discuss, as the use of set is a little more than a convenience factor here.

Reply:
hmm i think we would want to be able to remove rows from the page once the user selects all.  So for example, i select all, then remove 3. 

im not sure what you mean by `Maybe we would rather allow removal of sets from cohort and allow user to modify the sqon after the fact?`

Reply:
Thanks, I agree.
As for the other comment, it's basically enabling the "remove from cohort" button even if everything is selected (results will show 0), then user can still modify the sqon after the fact to get non-0 results. I don't think that's intuitive but it was an option so wanted your thoughts.

Reply:
Makes sense. I'll fix the comment on using `not-in` first and see if I can fix that one afterward.

Reply:
@hlminh2000 so in this scenario `then user can still modify the sqon after the fact to get non-0 results` would the user see the regular dropdown list of all the participant ids and then be able to select some ids?

Reply:
no, but that's a great point, the filter component doesn't know how to handle sets

Reply:
After changes to the way the sets and the sqon is built after removing participants, combining removed participants with other queries works nicely.
The way that the sets are represented in the cohort builder is still not super user-friendly, then.

Reply:
@jberube @hlminh2000 I would vote move forward so we can interact with this in the cohort builder and then we can adjust behaviour - is that possible?

Reply:
@hlminh2000 if you agree with @rosibaj , you know which button to push ;)